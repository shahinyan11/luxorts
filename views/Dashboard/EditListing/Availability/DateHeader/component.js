import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import classNames from 'classnames';

import checkDateInfo from 'utils/checkDateInfo';

const DateHeader = ({ label, date }) => {
  const { isActual, isToday } = checkDateInfo(date);

  return (
    <span
      className={classNames('booking-calendar__date', {
        'booking-calendar__date--overdue': !isActual,
      })}
    >
      {parseInt(label, 10)}
      {isToday && (
        <span className="d-none d-md-inline">
          {' '}
          <FormattedMessage id="shared.today" />
        </span>
      )}
    </span>
  );
};

DateHeader.propTypes = {
  label: PropTypes.string.isRequired,
  date: PropTypes.instanceOf(Date).isRequired,
};

export default DateHeader;
