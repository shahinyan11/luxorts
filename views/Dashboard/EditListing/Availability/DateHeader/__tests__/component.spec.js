import { shallow } from 'enzyme';

import DateHeader from '../component';

describe('DateHeader component', () => {
  const props = {
    date: new Date('15 Jun 2022 00:00:00 GMT'),
    label: '04',
  };

  const component = shallow(<DateHeader {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when date is actual', () => {
    component.setProps({
      ...props,
      date: new Date(),
    });

    expect(component).toMatchSnapshot();
  });
});
