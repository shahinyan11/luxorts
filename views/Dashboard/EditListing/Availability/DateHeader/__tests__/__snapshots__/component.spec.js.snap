// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateHeader component matches snapshot 1`] = `
<span
  className="booking-calendar__date booking-calendar__date--overdue"
>
  4
</span>
`;

exports[`DateHeader component matches snapshot when date is actual 1`] = `
<span
  className="booking-calendar__date"
>
  4
  <span
    className="d-none d-md-inline"
  >
     
    <Memo(FormattedMessage)
      id="shared.today"
    />
  </span>
</span>
`;
